modules/core/src/main/java/io/fluo/core/oracle/OracleServer.java

```diff
@@ -191,13 +195,25 @@ private boolean isLeader(Participant participant) {
   }
 
   @VisibleForTesting
+  public int getPort() {
+    return port;
+  }
+
+  @VisibleForTesting
   public boolean isConnected() {
     return (started && cnxnListener.isConnected());
   }
 
   private InetSocketAddress startServer() throws TTransportException {
 
-    InetSocketAddress addr = new InetSocketAddress(env.getConfiguration().getOraclePort());
+    if (env.getConfiguration().containsKey(FluoConfigurationImpl.ORACLE_PORT_PROP)) {
+      port = env.getConfiguration().getInt(FluoConfigurationImpl.ORACLE_PORT_PROP);
+      Preconditions.checkArgument(port >= 1 && port <= 65535,
+          FluoConfigurationImpl.ORACLE_PORT_PROP + " must be valid port (1-65535)");
+    } else {
+      port = PortUtils.getRandomFreePort();
+    }
```

Comment:
Offline, you mentioned logging a message w/ the oracle port.  Is that done anywhere?